scripts.js

```diff
@@ -32,39 +27,28 @@ document.getElementById('file-input').onchange = function (e) {
   );
 };
 
-
-
-
-// mainScript();
-
 function listenerStuff(curiosity) {
   const update = (event) => {
-    curiosity.setDivisions(xDivisionsSlider.value, yDivisionsSlider.value)
-    curiosity.setReductionRatio(ratioSlider.value/100);
-    curiosity.setOffset(xOffsetSlider.value/100, yOffsetSlider.value/100);
-
-    sliders.forEach((slider, index) => {
-      sliderOutputs[index].value = slider.value;
-    });
-
+    const percentageKeys = ['reductionRatio', 'xOffset', 'yOffset'];
+    const targetKey = event.target.dataset.key;
+    const sliderOutput = document.getElementById(`${event.target.id}-output`);
+    sliderOutput.value = event.target.value;
+
+    if (percentageKeys.includes( targetKey ) ) {
+      curiosity[targetKey] = event.target.value / 100;  
+    } else {
+      curiosity[targetKey] = event.target.value;  
+    }
+  
```

Comment:
Rewrite of this update function.

Smart usage of the "data-" attribute in HTML makes for easy updating curiosity values.

If statement required since sliders must be integers.